packages/@aws-cdk/aws-codebuild/lib/report-group.ts

```diff
@@ -0,0 +1,152 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { CfnReportGroup } from './codebuild.generated';
+import { renderReportGroupArn, reportGroupArnComponents } from './report-group-utils';
+
+/**
+ * The interface representing the ReportGroup resource -
+ * either an existing one, imported using the
+ * {@link ReportGroup.fromReportGroupName} method,
+ * or a new one, created with the {@link ReportGroup} class.
+ */
+export interface IReportGroup extends cdk.IResource {
+  /**
+   * The ARN of the ReportGroup.
+   *
+   * @attribute
+   */
+  readonly reportGroupArn: string;
+
+  /**
+   * The name of the ReportGroup.
+   *
+   * @attribute
+   */
+  readonly reportGroupName: string;
+
+  /**
+   * Grants the given entity permissions to write
+   * (that is, upload reports to)
+   * this report group.
+   */
+  grantWrite(identity: iam.IGrantable): iam.Grant;
+}
+
+abstract class ReportGroupBase extends cdk.Resource implements IReportGroup {
+  public abstract readonly reportGroupArn: string;
+  public abstract readonly reportGroupName: string;
+  protected abstract readonly exportBucket?: s3.IBucket;
+
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    const ret = iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [
+        'codebuild:CreateReport',
+        'codebuild:UpdateReport',
+        'codebuild:BatchPutTestCases',
+      ],
+      resourceArns: [this.reportGroupArn],
```

Comment:
Hm. Wonder if `CreateReport`/`UpdateReport` need a wildcard UNDER the report group, like log groups/log streams do (`${this.reportGroupArn}:*`). Could be that they do, could be that they don't.

Might be worth a quick confirmation?

Reply:
I can confirm the `*` is _not_ required: https://docs.aws.amazon.com/codebuild/latest/userguide/test-permissions.html

(I've also created report groups and projects that used them with these permissions, and everything worked correctly - I was able to see the uploaded reports after the builds finished)